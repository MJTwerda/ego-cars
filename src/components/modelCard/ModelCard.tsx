"use client";

import { VehicleModelI } from "./models.interfaces";
import Image from "next/image";
import { formatPrice } from "@/utils/numbers";

interface ModelListProps {
  vehicleData: VehicleModelI;
}

const ModelCard = ({ vehicleData }: ModelListProps) => {
  return (
    <div className="flex flex-col items-center justify-center p-4 border rounded-lg shadow-md 
                    transition-all duration-300 hover:scale-105 hover:shadow-lg hover:cursor-pointer">
      <h1 className="text-2xl font-semibold text-center">{vehicleData.name}</h1>

      <div className="flex items-center gap-2 text-gray-600">
        <p>{vehicleData.year}</p>
        <span>|</span>
        <p>${formatPrice(vehicleData.price)}</p>
      </div>

      <Image
        src={vehicleData.photo}
        alt={vehicleData.name}
        width={300}
        height={300}
        className="mt-4 rounded-md"
      />
    </div>
  );
};

export default ModelCard;

