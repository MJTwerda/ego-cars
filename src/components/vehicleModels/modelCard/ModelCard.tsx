"use client";

import { VehicleModelI } from "../interfaces/models.interfaces";
import Image from "next/image";
import Link from "next/link";

interface ModelListProps {
  vehicleData: VehicleModelI;
}

const ModelCard = ({ vehicleData }: ModelListProps) => {
  return (
    <div className="group hover:scale-105 transition-transform duration-300">
      <div className="flex flex-col items-center p-4">
        <div className="flex flex-col items-center">
          <h1 className="text-2xl font-semibold group-hover:text-[--color-active] min-h-[56px] text-center">
            {vehicleData.name}
          </h1>

          <div className="flex space-x-2 text-gray-600">
            <p>{vehicleData.year}</p>
            <p>|</p>
            <p>{vehicleData.price.toLocaleString("es-AR")}</p>
          </div>
        </div>

        <div className="flex justify-center items-center w-full h-[200px]">
          <Image
            src={vehicleData.photo}
            alt={vehicleData.name}
            width={300}
            height={200}
            className="rounded object-contain"
          />
        </div>

        <Link
          href={`/model-sheet?id=${vehicleData.id}`}
          className="bg-black text-white px-4 py-2 rounded-3xl opacity-0 group-hover:opacity-100 transition-opacity duration-300"
        >
          Ver Modelo
        </Link>
      </div>
    </div>
  );
};

export default ModelCard;
