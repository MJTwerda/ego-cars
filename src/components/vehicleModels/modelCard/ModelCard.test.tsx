import { act, render, RenderResult, waitFor } from '@testing-library/react';
import { useRouter } from 'next/navigation';
import ModelCard, { ModelListProps } from './ModelCard';

const mockProps: ModelListProps = {
  vehicleData: {
    id: 12345,
    segment: 'Segment 1',
    year: 2022,
    price: 10000,
    thumbnail: '/thumbnail1.jpg',
    name: 'Model 1',
    photo: '/img1.jpg',
  }
};

jest.mock('next/navigation', () => ({
  useRouter: jest.fn(),
}));

const mockPush = jest.fn();
(useRouter as jest.Mock).mockReturnValue({ push: mockPush });

describe('ModelCard', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render', async () => {
    let component: RenderResult;
    await act(async () => {
      component = render(<ModelCard vehicleData={mockProps.vehicleData} />);
    });

    await waitFor(() => {
      expect(component).toMatchSnapshot();
    });
  });

  it('should navigate when clicked', async () => {
    let component: RenderResult;
    await act(async () => {
      component = render(<ModelCard vehicleData={mockProps.vehicleData} />);
    });

    await waitFor(async () => {
      const link = component.getByRole('link');

      expect(link).toHaveAttribute('href', '/model-sheet?id=12345');
    });

  });
});