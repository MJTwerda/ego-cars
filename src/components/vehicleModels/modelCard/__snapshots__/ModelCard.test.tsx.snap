// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelCard should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="group hover:scale-105 transition-transform duration-300"
        href="/model-sheet?id=12345"
      >
        <div
          class="flex flex-col items-center p-4"
        >
          <div
            class="flex flex-col items-center"
          >
            <h1
              class="text-2xl font-semibold group-hover:text-[--color-active] min-h-[56px] text-center"
            >
              Model 1
            </h1>
            <div
              class="flex space-x-2 text-gray-600"
            >
              <p>
                2022
              </p>
              <p>
                |
              </p>
              <p>
                $ 
                10.000
              </p>
            </div>
          </div>
          <div
            class="flex justify-center items-center w-full h-[200px]"
          >
            <img
              alt="Model 1"
              class="rounded object-contain"
              data-nimg="1"
              decoding="async"
              height="200"
              loading="lazy"
              src="/_next/image?url=%2Fimg1.jpg&w=640&q=75"
              srcset="/_next/image?url=%2Fimg1.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg1.jpg&w=640&q=75 2x"
              style="color: transparent;"
              width="300"
            />
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary bg-black text-white px-4 py-2 rounded-3xl opacity-0 group-hover:opacity-100 transition-opacity duration-300 css-1uent87-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Ver Modelo
          </button>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="group hover:scale-105 transition-transform duration-300"
      href="/model-sheet?id=12345"
    >
      <div
        class="flex flex-col items-center p-4"
      >
        <div
          class="flex flex-col items-center"
        >
          <h1
            class="text-2xl font-semibold group-hover:text-[--color-active] min-h-[56px] text-center"
          >
            Model 1
          </h1>
          <div
            class="flex space-x-2 text-gray-600"
          >
            <p>
              2022
            </p>
            <p>
              |
            </p>
            <p>
              $ 
              10.000
            </p>
          </div>
        </div>
        <div
          class="flex justify-center items-center w-full h-[200px]"
        >
          <img
            alt="Model 1"
            class="rounded object-contain"
            data-nimg="1"
            decoding="async"
            height="200"
            loading="lazy"
            src="/_next/image?url=%2Fimg1.jpg&w=640&q=75"
            srcset="/_next/image?url=%2Fimg1.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg1.jpg&w=640&q=75 2x"
            style="color: transparent;"
            width="300"
          />
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary bg-black text-white px-4 py-2 rounded-3xl opacity-0 group-hover:opacity-100 transition-opacity duration-300 css-1uent87-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Ver Modelo
        </button>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
